python/ray/worker.py

```diff
@@ -686,13 +694,69 @@ def init(
             and the API is subject to change.
 
     Returns:
-        Address information about the started processes.
+        If the provided address included a protocol (e.g. "ray://1.2.3.4")
+        then a ClientContext is returned with information such as settings,
+        server versions for ray and python, and the dashboard_url.
+        Otherwise, returns address information about the started processes.
 
     Raises:
         Exception: An exception is raised if an inappropriate combination of
             arguments is passed in.
     """
 
+    # If available, use RAY_ADDRESS to override if the address was left
+    # unspecified, or set to "auto" in the call to init
+    address_env_var = os.environ.get(
+        ray_constants.RAY_ADDRESS_ENVIRONMENT_VARIABLE)
+    if address_env_var:
+        if address is None or address == "auto":
+            address = address_env_var
+            logger.info(
+                f"Using address {address_env_var} set in the environment "
+                f"variable {ray_constants.RAY_ADDRESS_ENVIRONMENT_VARIABLE}")
+
+    if address is not None and "://" in address:
+        # Address specified a protocol, use ray client
+        builder = ray.client(address)
+        builder._init_args(
```

Comment:
why do we need this?
I think I am missing something.

Reply:
The check for "://" in address is needed since the current plan is to make sure that init's behavior remains the same when the address doesn't have a protocol, which should keep us from breaking any existing code (existing code shouldn't have "someProtocol://" in the address, afaik)

The use of ray.client here is a bit conflicting since technically we would want to phase out ray.client, but it's currently where all the plugin logic (like loading a custom ClientBuilder) lives, and since we'll be keeping ClientBuilder working I figured it would be useful to just reuse that logic for now

Reply:
maybe a more proper way is to say:
`builder.store_args()`?